cmd/vicadmin/server.go

```diff
@@ -75,6 +77,80 @@ const (
 	genericErrorMessage    = "Internal Server Error; see /var/log/vic/vicadmin.log for details" // for http errors that shouldn't be displayed in the browser to the user
 )
 
+//wrapper struct around net.Conn for our custom funcs
+type Conn struct {
+	net.Conn
+	b             byte
+	err           error
+	oneTimeSwitch bool
+}
+
+//Returns number of bytes read
+func (c *Conn) Read(b []byte) (int, error) {
+	//oneTimeSwitch bool of first conn starts off as true so we do the TLS check once
+	if c.oneTimeSwitch {
```

Comment:
How about `c.UncertainTLS` for this flag (sorry to have you keep renaming it)? Then the code has some semantic meaning: "`if c.UncertainTLS` then determine whether TLS is present on this connection or not"

Hopefully you agree that makes the flag's purpose clearer?